source/common/stats/statsd.cc

```diff
@@ -114,10 +157,12 @@ void TcpStatsdSink::TlsSink::onEvent(uint32_t events) {
 
 void TcpStatsdSink::TlsSink::onTimespanComplete(const std::string& name,
                                                 std::chrono::milliseconds ms) {
-  write(fmt::format("envoy.{}:{}|ms\n", name, ms.count()));
+  // Ultimately it would be nice to perf optimize this path also, but it's not very frequent.
+  Buffer::OwnedImpl buffer(fmt::format("envoy.{}:{}|ms\n", name, ms.count()));
+  write(buffer);
```

Comment:
I assume there are no issues with ordering here, e.g. the timespan might appear before some other stats in output because it is committed to the buffer here, rather than in `endFlush()`.